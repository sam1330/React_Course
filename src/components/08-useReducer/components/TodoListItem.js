import React from "react";
import PropTypes from 'prop-types';

const TodoListItem = ({ todo, index, handleDelete, handleToggle }) => {

  return (
    <li
      key={todo.id}
      className={ todo.done ? 'done' : 'pending' }
    >
      <p
        onClick={() => handleToggle(todo.id)}
      >
        {index + 1}. {todo.desc}
      </p>
      <button
        className="btn btn-outline-danger ml-3"
        onClick={ () => handleDelete(todo.id) }
      >
        Delete
      </button>
    </li>
  )
};


TodoListItem.propTypes = {
  todo: PropTypes.object.isRequired,
  handleDelete: PropTypes.func,
  handleToggle: PropTypes.func
};

export default TodoListItem;