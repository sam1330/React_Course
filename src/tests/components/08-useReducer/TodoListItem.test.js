import { shallow } from "enzyme";
import React from "react";
import TodoListItem from "../../../components/08-useReducer/components/TodoListItem";
import { demoTodos } from "../../fixtures/demoTodos";


describe("<TodoListItem /> tests", () => {

  const handleDelete = jest.fn();
  const handleToggle = jest.fn();

  const wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      index={0}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  test('should render the component', () => {

    expect(wrapper).toMatchSnapshot();

  });

  test('should call handleDelete', () => {

    wrapper.find('button').simulate('click');
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);

  });
  
  test('should call handleToggle', () => {

    wrapper.find('p').simulate('click');
    expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);

  });
 
  test('should show the todo text', () => {

    const p = wrapper.find('p').text();

    expect(p).toBe(`1. ${demoTodos[0].desc.trim()}`)

  });

  test('should apply complete className', () => {
    
    const todo = demoTodos[0];
    todo.done = true;

    const wrapper = shallow(
      <TodoListItem
        todo={todo}
      />
    );

    expect(wrapper.find('li').hasClass('done')).toBe(true);

  });
});