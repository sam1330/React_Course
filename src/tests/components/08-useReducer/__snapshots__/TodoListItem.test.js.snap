// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoListItem /> tests should render the component 1`] = `
<li
  className="pending"
  key="1"
>
  <p
    onClick={[Function]}
  >
    1
    . 
    Learn React
  </p>
  <button
    className="btn btn-outline-danger ml-3"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
